src/Files.App/Helpers/DynamicDialogFactory.cs

```diff
@@ -57,50 +57,27 @@ public static DynamicDialog GetFor_ShortcutNotFound(string targetPath)
 
 		public static DynamicDialog GetFor_RenameDialog()
 		{
-			DynamicDialog dialog = null;
-			TextBox inputText = new TextBox()
+			DynamicDialog? dialog = null;
+			TextBox inputText = new()
 			{
 				Height = 35d,
 				PlaceholderText = "RenameDialogInputText/PlaceholderText".GetLocalizedResource()
 			};
 
-			TextBlock tipText = new TextBlock()
+			TextBlock tipText = new()
 			{
 				Text = "RenameDialogSymbolsTip/Text".GetLocalizedResource(),
 				Margin = new Microsoft.UI.Xaml.Thickness(0, 0, 4, 0),
 				TextWrapping = Microsoft.UI.Xaml.TextWrapping.Wrap,
 				Opacity = 0.0d
 			};
 
-			inputText.BeforeTextChanging += async (textBox, args) =>
+			inputText.TextChanged += (textBox, args) =>
 			{
-				if (FilesystemHelpers.ContainsRestrictedCharacters(args.NewText))
-				{
-					args.Cancel = true;
-					await inputText.DispatcherQueue.EnqueueAsync(() =>
-					{
-						var oldSelection = textBox.SelectionStart + textBox.SelectionLength;
-						var oldText = textBox.Text;
-						textBox.Text = FilesystemHelpers.FilterRestrictedCharacters(args.NewText);
-						textBox.SelectionStart = oldSelection + textBox.Text.Length - oldText.Length;
-						tipText.Opacity = 1.0d;
-					});
-				}
-				else
-				{
-					dialog.ViewModel.AdditionalData = args.NewText;
-
-					if (!string.IsNullOrWhiteSpace(args.NewText))
-					{
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
-					}
-					else
-					{
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Cancel;
-					}
-
-					tipText.Opacity = 0.0d;
-				}
+				tipText.Opacity = FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) ? 1.0d : 0.0d;
+				dialog!.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) || FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) 
```

Comment:
Shouldn't the test test also for `ContainsRestrictedFileName(name)` ?
If so, we could use here the `IsValidForFilename` method that is introduced in #10967 to simplify the test.

Reply:
I think we should merge #10967 first and then we can use the method

Reply:
@yaira2

Reply:
@hecksmosis your move to update and use the method now!

Reply:
Done